bin/wasm-node/javascript/src/worker/bindings-smoldot-light.ts

```diff
@@ -0,0 +1,310 @@
+                        const len = Buffer.byteLength(message, 'utf8');
+                        const ptr = instance.exports.alloc(len) >>> 0;
+                        Buffer.from(instance.exports.memory.buffer).write(message, ptr);
+                        instance.exports.connection_closed(id, ptr, len);
+                    };
+                    connection.socket.onmessage = (msg) => {
+                        const message = Buffer.from(msg.data as ArrayBuffer);
+                        const ptr = instance.exports.alloc(message.length) >>> 0;
+                        message.copy(Buffer.from(instance.exports.memory.buffer), ptr);
+                        instance.exports.connection_message(id, ptr, message.length);
+                    };
+
+                } else if (tcpParsed != null) {
+                    // `net` module will be missing when we're not in NodeJS.
+                    if (!compat.isTcpAvailable() || config.forbidTcp) {
+                        throw new ConnectionError('TCP connections not available');
+                    }
+
+                    const socket = compat.createConnection({
+                        host: tcpParsed[2],
+                        port: parseInt(tcpParsed[3] as string, 10),
+                    });
+
+                    connection = { ty: 'tcp', socket };
+                    connection.socket.setNoDelay();
+
+                    connection.socket.on('connect', () => {
+                        if (socket.destroyed) return;
+                        instance.exports.connection_open(id);
+                    });
+                    connection.socket.on('close', (hasError) => {
+                        if (socket.destroyed) return;
+                        // NodeJS doesn't provide a reason why the closing happened, but only
+                        // whether it was caused by an error.
+                        const message = hasError ? "Error" : "Closed gracefully";
+                        const len = Buffer.byteLength(message, 'utf8');
+                        const ptr = instance.exports.alloc(len) >>> 0;
+                        Buffer.from(instance.exports.memory.buffer).write(message, ptr);
+                        instance.exports.connection_closed(id, ptr, len);
+                    });
+                    connection.socket.on('error', () => { });
+                    connection.socket.on('data', (message) => {
+                        if (socket.destroyed) return;
+                        const ptr = instance.exports.alloc(message.length) >>> 0;
+                        message.copy(Buffer.from(instance.exports.memory.buffer), ptr);
+                        instance.exports.connection_message(id, ptr, message.length);
+                    });
+
+                } else {
+                    throw new ConnectionError('Unrecognized multiaddr format');
+                }
+
+                connections[id] = connection;
+                return 0;
+
+            } catch (error) {
```

Comment:
```suggestion
            } catch (error: Error | ConnectionError) {
```

Reply:
Doesn't compile. Catch clause can only be `any` or `unknown`